File: 0231-power-of-two/0231-power-of-two.java
//Iterative Approach
// class Solution {
//     public boolean isPowerOfTwo(int n) {
//        if (n <= 0) {
//             return false;
//         }
//         while (n % 2 == 0) {
//             n = n / 2;
//         }
//         return n == 1;
//     }
// }
// BIT MANIPULATION

class Solution {
    public boolean isPowerOfTwo(int n) {
        return n > 0 && (n & (n - 1)) == 0;
    }
}

File: 0231-power-of-two/0231-power-of-two.cpp
// class Solution {
// public:
//     bool isPowerOfTwo(int n) {
//         if(n==0){
//             return false;
//         }
//         while(n%2==0) n = n/2;
//         return n==1;       
//     }
// };
// USING RECURSION
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if(n==0) return false;
        return ((n==1) || (n%2==0 && isPowerOfTwo(n/2)));
    }
};